src/common-util/src/main/java/ee/ria/xroad/common/util/HttpSender.java

```diff
@@ -111,6 +112,11 @@ private void doRequest(HttpRequestBase request) throws Exception {
 
         addAdditionalHeaders();
 
+        log.debug("Log request headers");
```

Comment:
Maybe this should be trace-level logging? Unnecessary looping could be avoided by checking the enabled logging level (log.isTraceEnabled()).